rq/job.py

```diff
@@ -485,6 +485,8 @@ def delete(self, pipeline=None):
     # Job execution
     def perform(self):  # noqa
         """Invokes the job function with the job arguments."""
+        self.connection.persist(self.key)
+        self.ttl = self.connection.ttl(self.key)
```

Comment:
What's the intention of this line? Doesn't `connection.ttl` always return `-1` since we called `connection.persist` on the line above?